### ISSUE TITLE
How to do dialog discourse parser with my own data?

### ISSUE THREAD

**Issue by Hannibal046** [xcfcode/DDAMS#4]
Hello,
Thanks for the open source code!
I am wondering how can I use dialogdiscourse parser with my own dialog data? Should I put all training data in datadir/train.json?
I tried to make every training instance into a json file and put them together in train.json. Also, I add a list.my_data in the folder.
the dataformat of train.json is `List[Dict]`,with keys:`edus,id,relation(empty)`
But it seems that it did not work.

```python
Traceback (most recent call last):
  File "main.py", line 78, in <module>
    data_test = load_data("./" + FLAGS.data + "/" + FLAGS.test_data, map_relations)
  File "/home/mist/DialogueDiscourseParser/utils.py", line 21, in load_data
    for edu in dialog["edus"]:
TypeError: string indices must be integers

**Comment by xcfcode** [comment]
Thanks for your attention!!!
**The parser needs one JSON file for one meeting, rather than all meetings in one JSON file.**
one meeting should be formatted in the following form:
```
[
  {
    "edus": [
      {
        "text": "this is the kick-off meeting for our project .",
        "speaker": "B"
      },
      {
        "text": "and this is just what we're gonna be doing over the next twenty five minutes .",
        "speaker": "A"
      },
      ...
    ],
    "id": "xxxxx",
    "relations": [
      {
        "y": 1,
        "x": 0,
        "type": "Explanation"
      }
    ]
  }
]
```

**Comment by Hannibal046** [comment]
Thanks for the reply !
The scripts works perfectly for ICSI and AMI corpus.
But I still can not use this script on my own data. Even if I set the data format exactly as you said. The right is ICSI data, and left is my own data.
<img width="1845" alt="image" src="https://user-images.githubusercontent.com/38466901/121896800-485e0400-cd54-11eb-98da-00ce7a9c422f.png">

<img width="1811" alt="image" src="https://user-images.githubusercontent.com/38466901/121896914-69265980-cd54-11eb-8312-4b297b95483f.png">

And I change the `utils/load_data` 

from 

```python

    f_in = open(filename)
    inp = f_in.readline()
    data = json.loads(inp)
```
to

```python
    f_in = open(filename)
    data = json.load(f_in)
```

and here is the creepy mistake info from tensor flow
<img width="1060" alt="image" src="https://user-images.githubusercontent.com/38466901/121897661-30d34b00-cd55-11eb-881c-344aeeb8dd63.png">

And I have no idea what to do next. 
Appreciate your help!

**Comment by Hannibal046** [comment]
or Should I pretokenize my data with space ?

**Comment by xcfcode** [comment]
![image](https://user-images.githubusercontent.com/42307427/121977365-6bba9a80-cdb8-11eb-9f58-a4160fe4eb1b.png)
yes, my data is tokenized with space. have you tried this?

**Comment by Hannibal046** [comment]
Hello! 
I did not tokenized with space, but I think it will only matters when punctuation tightly follows a word, which I assume it will be treated like a \<unk> , causing lower performance.
I did some debugging and I found that all mistake info come from `/main.py/saver.restore(sess,tf.train.latest_checkpoint(FLAGS.model_dir))` . 
When  using AMI or ICSI corpus, everything works fine.
I did the following change to your code:

1. prepare the data as ICSI and AMI with a dummy  `train.json`  and  `test.json`
2. change `main.py` with `is_train=False`,`word_vector=my/path/to/glove`,`data=my/path/to/data`
3. generate a `list.my_data` and put it in the `run_test.py`
4. change the output_dir in `utils.py`

Here is the mistake info
<img width="1059" alt="image" src="https://user-images.githubusercontent.com/38466901/122023214-e8ba3400-cdf9-11eb-9ab9-c335b0f55e55.png">

It says ` Assign requires shapes of both tensors to match. lhs shape= [3,100] rhs shape= [2984,100]
`,100 maybe the glove vector. I don't know what the others terms mean.

Thanks !

**Comment by Hannibal046** [comment]
I notice that all your edu is a single sentence with only one terminating punctuation . And it is all relatively short compared to my data. Does this have anything to do with the model, like truncation or padding ?

**Comment by xcfcode** [comment]
Hi, are your currently satisfy the following conditions?
1. You can first copy `train.json` and `test.json` under the AMI directory to your own data directory **without any modification**, results in `yourdata/train.json` and `yourdata/test.json` since these two files are only for code running rather than discourse parsing.
2. Then, for your meetings, each meeting is converted into a JSON file, namely, `meetingX.json` under the `yourdata` directory. 
3. Finally, you may have a data directory `yourdata` including `meeting1.json`, `meeting2.json`, ..., `train.json`, and `test.json`.
4. **Please note that `train.json`, and `test.json` are directly copied from AMI directory.**

**Comment by Hannibal046** [comment]
I got it ! 
Previously, I executed following command:
1. `cp random_meeting_data.json train.json`
2. `touch test.json`

Since they are not important in the inference phase.

Then I just did another try to directly copy `AMI/train.json` to my data folder. And it works! I rushed to here to tell you, but found out that you already give the correct answer. 

Thanks so much ! Appreciate your work on dialog summarization. 
By the way, the slides you made about dialog summarization is excellent !
